src/components/Card.js

```diff
@@ -1,29 +1,53 @@
 export default class Card {
-  constructor(data, cardSelector, handleImageClick) {
-    this._name = data.name;
-    this._link = data.link;
+  constructor(data, cardSelector, handleImageClick, handleDeleteClick,
+    handleLikeClick) {
+    this.name = data.name;
+    this.link = data.link; 
+    this._id = data._id; 
+    this.isLiked = data.isLiked;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
-  }
+    this._handleLikeClick = handleLikeClick;
+    this._handleDeleteClick = handleDeleteClick;
+  } 
 
+  getCardId() {
+    return this._id;
+  }
+  
   _setEventListeners() {
     this._likeButton.addEventListener("click", () => {
-      this._handleLikeIcon();
+      this.handleLikeIcon(this);
     });
+
     this._deleteButton.addEventListener("click", () => {
-      this._handleDeleteCard();
+      this.handleDeleteCard(this);
     });
 
+    // this._cardElement
+    // .querySelector(".delete__button")
+    // .addEventListener("click", () => {
+    //   this._handleDeleteClick(this);
+    // });
+
     this._cardImageEl.addEventListener("click", () => {
-      this._handleImageClick({name: this._name, link: this._link});
+      this._handleImageClick({name: this.name, link: this.link});
     });
   }
 
-  _handleLikeIcon() {
+  handleLikeIcon() {
```

Comment:
Needs correction: please remove this method, you don't need it anymore. You should create a setter for the likes instead:

```js
// call this method from index.js after successful api response in `then` block
setIsLiked(isLiked) { 
   // set instance variable
    this._isLiked = isLiked; 
    this._renderLikes(); 
} 
```